### PR TITLE
Ab/p2022 1078 status controller

### PR DESCRIPTION
**Pull request by Papuzka**
Status Controller:

- Get all, get by id, create, update, delete status actions,

Using query as an input parameter instead of DTO object, hence no fluent validation usage (although one validation class is added in case the need for the approach to be changed) simple validation is done in StatusService
Any suggestions appreciated

### CODE REVIEW

File: src/Patronage.Api/Controllers/StatusController.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                    ResponseCode = StatusCodes.Status201Created,
+                    Data = (int)id,
+                    Message = "Status created successfully"
+                });
+            }            
+            return BadRequest(new BaseResponse<string>
+            {
+                ResponseCode = StatusCodes.Status400BadRequest,
+                Data = null,
+                Message = "Status code already exists"
```
Comment (mateuszgiedzinski) [inline src/Patronage.Api/Controllers/StatusController.cs:83]: Is this validation error? Do we use FluentValidation for such cases? I think this should be app error

File: src/Patronage.Api/Controllers/StatusController.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            var status = await _mediator.Send(new UpdateStatusCommand(id, code));
+            return status;
+        }
+        [SwaggerOperation(Summary = "Delete Status by id", Description = "Delete Status specifying statusId")]
+        [HttpDelete]
+        [SwaggerResponse(StatusCodes.Status200OK, "Resource deleted successfully")]
+        [SwaggerResponse(StatusCodes.Status400BadRequest, "An error occured trying to delete resource")]
+        public async Task<ActionResult<bool>> Delete([FromQuery] int id)
+        {
+            var deleted = await _mediator.Send(new DeleteStatusCommand(id));
```
Comment (mateuszgiedzinski) [inline src/Patronage.Api/Controllers/StatusController.cs:98]: name this isSucceeded or succeeded. I prefere naming with prefix is  for boolean.

File: src/Patronage.DataAccess/Services/StatusService.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        private readonly TableContext _dbContext;
+
+        public StatusService(TableContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+        public async Task<IEnumerable<StatusDto>> GetAll()
+        {
+            var statusesQueryable = _dbContext
```
Comment (mateuszgiedzinski) [inline src/Patronage.DataAccess/Services/StatusService.cs:18]: why you split to statusQueryable before executing Select ? 
We don't need executing Asqueryable. Select should work od DbSet becouse DbSet implements Iqueryable interface.

File: src/Patronage.DataAccess/Services/StatusService.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            var statuses = await statusesQueryable
+                          .Select(status => new StatusDto
+                          {
+                              Id = status.Id,
+                              Code = status.Code
+                          })
+                          .ToListAsync();
+            return statuses;
+        }
+        public async Task<StatusDto?> GetById(int id)
```
Comment (mateuszgiedzinski) [inline src/Patronage.DataAccess/Services/StatusService.cs:31]: Warnings for nullability

File: src/Patronage.DataAccess/Services/StatusService.cs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+            await _dbContext.SaveChangesAsync();
+            return status.Id;
+
+
+        }
+        public async Task<bool> Delete(int statusId)
+        {
+            var status = await _dbContext.Statuses
+                        .FirstOrDefaultAsync(b => b.Id == statusId);
+            _dbContext.Statuses.Remove(status);
```
Comment (mateuszgiedzinski) [inline src/Patronage.DataAccess/Services/StatusService.cs:71]: warnings

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: src/Patronage.Api/Controllers/StatusController.cs (hunk 1 above)
Comment (Papuzka) [reply src/Patronage.Api/Controllers/StatusController.cs:83]: I could use fluent validation if I used DTO object to pass the string code. Instead the validation is done in service and data is passed from query. Should i change that to DTO? As a reminder I'd say its just one property.

File: src/Patronage.Api/Controllers/StatusController.cs (hunk 2 above)
Comment (mateuszgiedzinski) [reply src/Patronage.Api/Controllers/StatusController.cs:98]: I mean isSucceeded for all commands. If command ends successfully it returns boolean. I think it also should be named isSucceeded.

**Pull request merged**